hycom/RELO/src_2.2.98ZA-07Tsig0-i-sm-sse_relo_mpi/mod_hycom_fabm.F90

```diff
@@ -887,13 +1025,77 @@ subroutine hycom_fabm_update(m, n, ibio)
         input => input%next
       end do
 !write (*,*) 'hycom_fabm_after_correct', nstep, time
+          do ivar=1,size(fabm_model%state_variables)
+           if (trim(fabm_model%state_variables(ivar)%name)=='ECO_opa') then
+             do i=1,ii
+              do j=1,jj
+                 if (i==itest .and. j==jtest) then
+                 mass1Dafter(:,ivar) = tracer(i, j, :, n, ivar) * dp(i, j, : ,n)/onem
+                 write(*,*)'INTERIOR MASS DIFF chk,',sum(mass1Dafter(1:kbottom(i,j,n),ivar)) - sum(mass1Dbefore(1:kbottom(i,j,n),ivar))
+                 write(*,*)'INTERIOR after chk,',tracer(i,j,kbottom(i,j,n)-1:kbottom(i,j,n)+1,n, ivar),kbottom(i,j,n)
+                 write(*,*)'INTERIOR after thk chk,',dp(i,j,kbottom(i,j,n)-1:kbottom(i,j,n)+1,n)/onem,kbottom(i,j,n)
+                 write(*,'(A25,2x,F14.4,2x,F14.4,2x,I02,2x,F7.3)')'CHECK after interior chk',sum(tracer(i, j, :, m, ivar) * dp(i, j, : , m)/onem), sum(tracer(i, j,:, n, ivar) * dp(i, j, : , n)/onem),modelmonth,modelday
+                 write(*,*)'CHECK ---------'
+                 endif
+              enddo
+             enddo
+           endif
+          enddo
+
+          do i=1,ii
+            do j=1,jj
+              do ivar=1,size(fabm_model%state_variables)
+                mass_before_check_state(i, j, :, ivar) = tracer(i, j, :, n, ivar) * dp(i, j, :, n)/onem
```

Comment:
@caglar, small thing: you could save the space taken up by "mass_before_check" state by computing the sum in this step - then you can remove the depth dimension.